cmd/influx/write.go

```diff
@@ -68,94 +75,128 @@ or add an entire file specified with an @ prefix.`
 		},
 	}
 	opts.mustRegister(cmd)
+	cmd.Flags().StringVar(&writeFlags.Format, "format", "", "Input format, either lp (Line Protocol) or csv (Comma Separated Values). Defaults to lp unless '.csv' extension")
+	cmd.Flags().StringVarP(&writeFlags.File, "file", "f", "", "The path to the file to import")
+	cmd.Flags().BoolVar(&writeFlags.DryRun, "dry-run", false, "Write protocol lines to stdout")
 
 	return cmd
 }
 
 func fluxWriteF(cmd *cobra.Command, args []string) error {
 	ctx := context.Background()
-
-	if writeFlags.Org != "" && writeFlags.OrgID != "" {
-		return fmt.Errorf("please specify one of org or org-id")
+	if writeFlags.DryRun {
+		flags.Host = "-"
 	}
 
-	if writeFlags.Bucket != "" && writeFlags.BucketID != "" {
-		return fmt.Errorf("please specify one of bucket or bucket-id")
-	}
+	var bucketID, orgID platform.ID
+	// validate flags unless writing to stdout
+	if flags.Host != "-" {
+		if writeFlags.Org != "" && writeFlags.OrgID != "" {
+			return fmt.Errorf("please specify one of org or org-id")
+		}
 
-	if !models.ValidPrecision(writeFlags.Precision) {
-		return fmt.Errorf("invalid precision")
-	}
+		if writeFlags.Bucket != "" && writeFlags.BucketID != "" {
+			return fmt.Errorf("please specify one of bucket or bucket-id")
+		}
 
-	bs, err := newBucketService()
-	if err != nil {
-		return err
-	}
+		if !models.ValidPrecision(writeFlags.Precision) {
+			return fmt.Errorf("invalid precision")
+		}
 
-	var filter platform.BucketFilter
-	if writeFlags.BucketID != "" {
-		filter.ID, err = platform.IDFromString(writeFlags.BucketID)
+		bs, err := newBucketService()
```

Comment:
If you flag dry host to be '-', you could get an error here, since bucket service is a tcp request to the host.